### ISSUE TITLE
Error is undefined on the server when a variable with a default value is not set in `variables`

### ISSUE THREAD

**Issue by Dattaya** [apollographql/apollo-client#11090]
### Issue Description

I'm experiencing this bug starting from 3.5.10, it's not a big deal, because this problem is easily fixable. Maybe it's not even classified as a bug, but this behavior is confusing IMO. 
It might've been reported before, because some people are mentioning 3.5.10 in #8063, maybe they didn't know what was causing the issue. 
Here's an example of a query that would get you in trouble, ie `error` from `useQuery` would be empty on the server:
```ts
// Erroring query with an unexpected `id`
const testQuery = gql`
  query TestVar($isEditable: Boolean = false) {
    artwork(id: "test-id") {
      slug
      internalID @include(if: $isEditable)
    }
  }  
`
const { error } = useQuery(testQuery) // undefined
```

### Link to Reproduction

https://github.com/Dattaya/ts-react-ssr-basis/tree/apollo-ssr-error-def-var

### Reproduction Steps

I disabled client-side react, so you don't have to open page's source to view the output from the server

* `git clone -b apollo-ssr-error-def-var --single-branch git@github.com:Dattaya/ts-react-ssr-basis.git`
* `cd ts-react-ssr-basis`
* `npm start`
* Open http://localhost:3000/test and you'll see the error
* Open http://localhost:3000/test-var and there will be no error, adding `isEditable` to variables makes it work again

**Comment by Dattaya** [comment]
I thought it was my only issue, but it turns out there's another more severe bug, and nothing helps other than making it a required variable: `$isEditable: Boolean!` and removing the default. But I don't have enough time to figure out how to reproduce it.

**Comment by bignimbus** [comment]
Hi @Dattaya 👋🏻 thanks for opening this issue!  FWIW, the `if` variable passed to `@include` is required per [the spec](https://spec.graphql.org/draft/#sec--include):

```graphql
directive @include(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
```

So it makes sense that annotating `$isEditable: Boolean!` would make a difference.  You can still set a default for non-nullable variables in GraphQL, though:

```graphql
$isEditable: Boolean! = false
```

I'll plan to pull down your reproduction and test a few things later on but hopefully this helps clarify things until then.

**Comment by Dattaya** [comment]
yep, this clarifies a lot, thanks!

> the `if` variable passed to `@include` is required per [the spec](https://spec.graphql.org/draft/#sec--include):

I had no idea, so it seems like it’s my misunderstanding of the graphql spec rather than Apollo’s problem. If you want and it’s possible, adding some warning/error would be very helpful for the users that misuse the library.

**Comment by bignimbus** [comment]
Thanks for the feedback - if you're using TypeScript with GraphQL Code Generator this should be able to be caught at compile time.  As for the undefined error, I'm trying to see what your server is returning but am not able to get the reproduction to run with the steps given.  I'm stuck on `98% emitting after emit webpack-shell-plugin-nextExecuting post-build scripts`.

Judging by the fact that the `server` directory doesn't use Apollo Server and there's no reference to the `backend` directory, which does use Apollo Server, maybe there's a connection there?  Not sure, tough to say without running the app.  I'm going to close this issue as completed but please do feel free to let us know for certain if you feel like validation errors properly being reported by a GraphQL server aren't being handled by Apollo Client as expected.  Thanks!

**Comment by Dattaya** [comment]
I'm sorry, I only tried to run this repro on Ubuntu and on other OSes it might not work at all. Apollo is connected in `server` in [`render.tsx`](https://github.com/Dattaya/ts-react-ssr-basis/blob/apollo-ssr-error-def-var/server/render.tsx#L52), as for `backend`, I used it to test other things, but it's not necessary here, so the app [connects to `https://metaphysics-production.artsy.net/v2`](https://github.com/Dattaya/ts-react-ssr-basis/blob/apollo-ssr-error-def-var/common/config.ts#L2C18-L2C61) instead.

So I tried to make the `$isEditable` variable required, unfortunately, it didn't help. But it's a minor issue, so I have nothing against your decision to close it.
